src/main/java/software/amazon/cloudwatchlogs/emf/util/Validator.java

```diff
@@ -88,10 +93,13 @@ public static void validateDimensionSet(String dimensionName, String dimensionVa
      * @param name Metric name
      * @param value Metric value
      * @param unit Metric unit
+     * @param storageResolution Metric resolution
      * @throws InvalidMetricException if metric is invalid
      */
-    public static void validateMetric(String name, double value, Unit unit)
+    public static void validateMetric(
+            String name, double value, Unit unit, StorageResolution storageResolution)
```

Comment:
As pointed in this [review](https://github.com/awslabs/aws-embedded-metrics-java/pull/129/files?short_path=b335630#r1054445785), we can pass additional map parameter from MetricContext class for just validation. We shouldn't be persisting data on Validator class which is meant for only validation purpose.

Reply:
Updated!